File: _salvage_from_refs/pull/1085/head/packages/sdk-ts/examples/tasks/http-get.ts
import { defineTask, createRunContext, type TaskInput } from ' @summit/maestro-sdk';

export default defineTask<{ url: string }, { status: number; body: string }> ({
  async validate(input: TaskInput<{ url: string }>) {
    if (!input.payload?.url) throw new Error('url is required');
  },
  async execute(ctx, input) {
    const res = await fetch(input.payload.url);
    const body = await res.text();
    ctx.logger.info('http-get', { status: res.status, bytes: body.length });
    await ctx.emit('http_get.done', { status: res.status });
    return { payload: { status: res.status, body } };
  }
});

// Local demo
if (process.env.NODE_ENV === 'development') {
  const ctx = createRunContext({});
  const task = (await import('./http-get.ts')).default;
  task.execute(ctx, { payload: { url: 'https://example.com' } }).then(r => console.log(r));
}

File: packages/sdk-py/examples/http_get.py
import requests
from maestro_sdk.task import define_task
from maestro_sdk.types import RunContext

http_get = define_task(
    {
        "validate": lambda input: (
            (_ for _ in ()).throw(Exception("url required")) if not input.get("url") else None
        ),
        "execute": lambda ctx, input: {"status": requests.get(input["url"]).status_code},
    }
)

if __name__ == "__main__":
    ctx = RunContext()
    print(http_get.execute(ctx, {"url": "https://example.com"}))
